--- a/avahi-ui/Makefile.am	2010-06-29 04:30:35.000000000 +0100
+++ b/avahi-ui/Makefile.am	2011-09-28 15:24:29.000000000 +0100
@@ -15,7 +15,7 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
 # USA.
 
-AM_CFLAGS=-I$(top_srcdir) -DG_DISABLE_DEPRECATED=1 -DGDK_DISABLE_DEPRECATED=1 -DGTK_DISABLE_DEPRECATED=1
+AM_CFLAGS=-I$(top_srcdir) -DGDK_DISABLE_DEPRECATED=1 -DGTK_DISABLE_DEPRECATED=1
 
 # This cool debug trap works on i386/gcc only
 AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
--- a/avahi-ui/Makefile.in	2011-04-03 23:10:46.000000000 +0100
+++ b/avahi-ui/Makefile.in	2011-09-28 15:24:46.000000000 +0100
@@ -458,7 +458,7 @@
 top_srcdir = @top_srcdir@
 
 # This cool debug trap works on i386/gcc only
-AM_CFLAGS = -I$(top_srcdir) -DG_DISABLE_DEPRECATED=1 \
+AM_CFLAGS = -I$(top_srcdir) \
 	-DGDK_DISABLE_DEPRECATED=1 -DGTK_DISABLE_DEPRECATED=1 \
 	'-DDEBUG_TRAP=__asm__("int $$3")' $(am__append_1)
 desktopdir = $(datadir)/applications
