diff -uNr NVIDIA-Linux-x86-96.43.11-pkg0.p/usr/src/nv/nv.c NVIDIA-Linux-x86-96.43.11-pkg0/usr/src/nv/nv.c
--- NVIDIA-Linux-x86-96.43.11-pkg0.p/usr/src/nv/nv.c	2009-02-24 00:46:00.000000000 +0100
+++ NVIDIA-Linux-x86-96.43.11-pkg0/usr/src/nv/nv.c	2009-06-19 21:47:50.000000000 +0200
@@ -499,9 +499,9 @@
      * Set the module owner to ensure that the reference
      * count reflects accesses to the proc files.
      */
-    proc_nvidia->owner = THIS_MODULE;
-    proc_nvidia_cards->owner = THIS_MODULE;
-    proc_nvidia_warnings->owner = THIS_MODULE;
+//    proc_nvidia->owner = THIS_MODULE;
+//    proc_nvidia_cards->owner = THIS_MODULE;
+//    proc_nvidia_warnings->owner = THIS_MODULE;
 
     for (j = 0; j < num_nv_devices; j++)
     {
@@ -521,7 +521,7 @@
 
         entry->data = nv;
         entry->read_proc = nv_kern_read_cardinfo;
-        entry->owner = THIS_MODULE;
+//        entry->owner = THIS_MODULE;
 
         if (nvos_find_agp_capability(dev)) {
             /*
@@ -534,7 +534,7 @@
                 goto failed;
             }
 
-            entry->owner = THIS_MODULE;
+//            entry->owner = THIS_MODULE;
             proc_nvidia_agp = entry;
 
             entry = create_proc_entry("status", flags, proc_nvidia_agp);
@@ -545,7 +545,7 @@
 
             entry->data = nv;
             entry->read_proc = nv_kern_read_status;
-            entry->owner = THIS_MODULE;
+//            entry->owner = THIS_MODULE;
 
             entry = create_proc_entry("host-bridge", flags, proc_nvidia_agp);
             if (!entry) {
@@ -555,7 +555,7 @@
 
             entry->data = NULL;
             entry->read_proc = nv_kern_read_agpinfo;
-            entry->owner = THIS_MODULE;
+//            entry->owner = THIS_MODULE;
 
             entry = create_proc_entry("card", flags, proc_nvidia_agp);
             if (!entry) {
@@ -565,7 +565,7 @@
 
             entry->data = nv;
             entry->read_proc = nv_kern_read_agpinfo;
-            entry->owner = THIS_MODULE;
+//            entry->owner = THIS_MODULE;
         }
 
         NV_PCI_DEV_PUT(dev);
@@ -576,14 +576,14 @@
         goto failed;
 
     entry->read_proc = nv_kern_read_version;
-    entry->owner = THIS_MODULE;
+//    entry->owner = THIS_MODULE;
 
     entry = create_proc_entry("registry", flags, proc_nvidia);
     if (!entry)
         goto failed;
 
     entry->read_proc = nv_kern_read_registry;
-    entry->owner = THIS_MODULE;
+//    entry->owner = THIS_MODULE;
 
     return;
 
@@ -610,7 +610,7 @@
 
     entry->data = (void *)message;
     entry->read_proc = nv_kern_read_warning;
-    entry->owner = THIS_MODULE;
+//    entry->owner = THIS_MODULE;
 #endif
 }
 
