diff -uNr gwhere-0.2.3.org/intl/Makefile.in gwhere-0.2.3/intl/Makefile.in
--- gwhere-0.2.3.org/intl/Makefile.in	2004-03-09 00:44:17.000000000 +0100
+++ gwhere-0.2.3/intl/Makefile.in	2011-10-13 19:29:59.000000000 +0200
@@ -205,7 +205,7 @@
 @GT_YES@intlh.inst: intlh.inst.in ../config.status
 @GT_YES@	cd .. \
 @GT_YES@	&& CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= \
-@GT_YES@	  $(SHELL) ./config.status
+@GT_YES@	  $(SHELL) ./config.status \
 @GT_NO@.PHONY: intlh.inst
 @GT_NO@intlh.inst:
 
